src/app/branded/branded-checkout.component.js

```diff
@@ -78,17 +79,43 @@ class BrandedCheckoutController {
         this.checkoutStep = 'thankYou'
         break
     }
-    this.$window.document.querySelector('branded-checkout').scrollIntoView({ behavior: 'smooth' })
+    this.$element.querySelector('branded-checkout').scrollIntoView({ behavior: 'smooth' })
   }
 
-  previous () {
-    switch (this.checkoutStep) {
-      case 'review':
-        this.fireAnalyticsEvents('contact', 'payment')
-        this.checkoutStep = 'giftContactPayment'
-        break
+  previous (newStep) {
+    let scrollElement = 'branded-checkout'
+
+    if (this.checkoutStep === 'review') {
+      this.fireAnalyticsEvents('contact', 'payment')
+      this.checkoutStep = 'giftContactPayment'
+
+      switch (newStep) {
+        case 'contact':
+          scrollElement = 'contact-info'
+          break
+        case 'cart':
+          scrollElement = 'product-config-form'
+          break
+        case 'payment':
+          scrollElement = 'checkout-existing-payment-methods'
+          break
+      }
     }
-    this.$window.document.querySelector('branded-checkout').scrollIntoView({ behavior: 'smooth' })
+
+    // Watch for changes until the element we are scrolling to exists and everything has loaded
+    // because there will be layout shift every time a new component finishes loading
+    const observer = new this.$window.MutationObserver(() => {
+      // TODO: When Firefox supports :has(), this query can be changed to `.panel :has(${scrollElement})`
```

Comment:
Note that give site must support older versions of browsers also, so we need to be careful using anything that is not supported by like 99% of browsers (we had to support IE 8 up until 2016, for example).

Reply:
Good point. I updated the comment.